modules/spring-config/src/main/java/org/mule/runtime/config/internal/registry/AbstractSpringRegistry.java

```diff
@@ -62,6 +64,8 @@ public abstract class AbstractSpringRegistry extends AbstractRegistry implements
   private RegistrationDelegate registrationDelegate;
   private boolean readOnly;
 
+  private final ReentrantLock stopOrDisposeLock = new ReentrantLock();
```

Comment:
why do you need a lock, instead of just a flag? What's the point of allowing objects to be registered after context stop instead of just throwing an exception?

Reply:
Original code was added to make a test pass. The flow under test was registering a ConfigurationProvider as part of the first event processing at the AsyncMessageProcessor. That was making the chance of "stopping while trying to add new entry" possible. Therefore I think we can safely fail but I will try to add also an integration test that nowadays can be achieved through the use of subflows (the original problem is no longer there because of several refactors at the ComponentMessageProcessor).